test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -1,7 +1,261 @@
 package e2e_sno_disruptive
 
-import "testing"
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"testing"
+	"time"
 
-func TestFallback(t *testing.T) {
-	t.Log("implement me")
+	"github.com/stretchr/testify/require"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/staticpod/startupmonitor/annotations"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	commontesthelpers "github.com/openshift/library-go/test/library/encryption"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+)
+
+func TestFallback(tt *testing.T) {
+	t := commontesthelpers.NewE(tt)
+	cs := getClients(t)
+	t.Log("Starting the fallback test")
+
+	// before starting a new test make sure the current state of the cluster is good
+	ensureClusterInGoodState(t, cs, 10*time.Second, 6*time.Minute, 1*time.Minute)
+
+	// cause a disruption
+	data := getDefaultUnsupportedConfig(t, cs)
+	//data["apiServerArguments"] = map[string]string{"non-existing-flag": "true"}
+	setUnsupportedConfig(t, cs, data)
+
+	// validate if the fallback condition is reported and the cluster is stable
+	waitForFallbackDegradedCondition(t, cs)
+	nodeName, failedRevision := assertNodeStatus(t, cs)
+	assertKasPodAnnotatedOnNode(t, cs, failedRevision, "*", "*", nodeName)
+	// TODO: try kas - get pod/ns ?
+	// TODO: check if pod is ready ?
+
+	// clean up
+	setUnsupportedConfig(t, cs, getDefaultUnsupportedConfig(t, cs))
+	if err := clusterInGoodState(t, cs, 10*time.Second, 6*time.Minute, 1*time.Minute); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func ensureClusterInGoodState(t testing.TB, cs clientSet, waitPollInterval, waitPollTimeout, mustBeReadyFor time.Duration) {
```

Comment:
does it need waitPollInterval ?

Reply:
do you want to use `wait.UntilWithContext` ?

Reply:
no, would just make this constant. 10s or 30s.